sdk/core/azure-core/src/main/java/com/azure/core/client/TokenCredentialSupport.java

```diff
@@ -0,0 +1,20 @@
+package com.azure.core.client;
```

Comment:
What is the right package for this to live in?

- Should we create new top level package `com.azure.core.client` for stuff related to clients and builders?
- Should we put this under `com.azure.core.utils` ?
- Should we nest it in case `client.x` grows ?

The argument against putting this under `utils` is that each `XClientBuilder` now `IS-A` `XSupport` so `XClientBuilder` `IS-AN util` would look not right.

The argument against top level package is that it might be overkill for just few micro interfaces. (Same argument applies to nesting).

Not nesting on the other hand means that if `client` (or `util.client`) package grows it becomes unorganized mess.

I'd name it `com.azure.core.client.traits` .

@JonathanGiles @lmolkova @srnagar @alzimmermsft Please share your thoughts.

Reply:
Let's go for `com.azure.core.client.traits` for now